paymentsheet/src/test/java/com/stripe/android/paymentelement/confirmation/epms/ExternalPaymentMethodConfirmationDefinitionTest.kt

```diff
@@ -0,0 +1,257 @@
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.core.strings.resolvableString
+import com.stripe.android.isInstanceOf
+import com.stripe.android.model.Address
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.paymentelement.confirmation.ConfirmationDefinition
+import com.stripe.android.paymentelement.confirmation.ConfirmationHandler
+import com.stripe.android.paymentelement.confirmation.FakeConfirmationOption
+import com.stripe.android.paymentelement.confirmation.asCanceled
+import com.stripe.android.paymentelement.confirmation.asFail
+import com.stripe.android.paymentelement.confirmation.asFailed
+import com.stripe.android.paymentelement.confirmation.asLaunch
+import com.stripe.android.paymentelement.confirmation.asSucceeded
+import com.stripe.android.payments.core.analytics.ErrorReporter
+import com.stripe.android.payments.paymentlauncher.PaymentResult
+import com.stripe.android.paymentsheet.ExternalPaymentMethodConfirmHandler
+import com.stripe.android.paymentsheet.ExternalPaymentMethodContract
+import com.stripe.android.paymentsheet.ExternalPaymentMethodInput
+import com.stripe.android.paymentsheet.ExternalPaymentMethodInterceptor
+import com.stripe.android.paymentsheet.R
+import com.stripe.android.testing.FakeErrorReporter
+import com.stripe.android.testing.PaymentIntentFactory
+import com.stripe.android.utils.DummyActivityResultCaller
+import com.stripe.android.utils.FakeActivityResultLauncher
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+
+class ExternalPaymentMethodConfirmationDefinitionTest {
+    @Test
+    fun `'key' should be 'ExternalPaymentMethod`() {
+        val definition = createExternalPaymentMethodConfirmationDefinition()
+
+        assertThat(definition.key).isEqualTo("ExternalPaymentMethod")
+    }
+
+    @Test
+    fun `'option' return casted 'ExternalPaymentMethod'`() {
+        val definition = createExternalPaymentMethodConfirmationDefinition()
+
+        assertThat(definition.option(EPM_CONFIRMATION_OPTION)).isEqualTo(EPM_CONFIRMATION_OPTION)
+    }
+
+    @Test
+    fun `'option' return null for unknown option`() {
+        val definition = createExternalPaymentMethodConfirmationDefinition()
+
+        assertThat(definition.option(FakeConfirmationOption())).isNull()
+    }
+
+    @Test
+    fun `'createLauncher' should register launcher properly for activity result`() = runTest {
+        val errorReporter = FakeErrorReporter()
+        val definition = createExternalPaymentMethodConfirmationDefinition(errorReporter)
+
+        var onResultCalled = false
+        val onResult: (PaymentResult) -> Unit = { onResultCalled = true }
+        val activityResultCaller = DummyActivityResultCaller()
+
+        definition.createLauncher(
+            activityResultCaller = activityResultCaller,
+            onResult = onResult,
+        )
+
+        val call = activityResultCaller.calls.awaitItem()
```

Comment:
Probably best as a follow up. But it seems like it would be good to update this to work differently. 

Ideally we'd pass a block, and during that we'd have access to the turbine, and when the block is over, we'd assert we consumed everything. Happy to pair, or take this on myself if you'd like.